include/openssl/ssl.h

```diff
@@ -1838,6 +1843,12 @@ OPENSSL_EXPORT const SSL_CIPHER *SSL_get_current_cipher(const SSL *ssl);
 // performed a new handshake.
 OPENSSL_EXPORT STACK_OF(SSL_CIPHER) *SSL_get_client_ciphers(const SSL *ssl);
 
+// SSL_client_hello_get0_ciphers provides access to the client ciphers field from the
+// Client Hello, optionally writing the result to an out pointer. It returns the field
+// length if successful, or 0 if |ssl| is a client or the handshake hasn't occurred yet.
+// Free |*out| after use if provided to the function.
```

Comment:
Should out be freed? Isn't it a pointer to the array in the SSL struct which will be cleaned up automatically?

Reply:
fixed, from previous implementation of the function